fedimint-api-client/src/api/mod.rs

```diff
@@ -919,123 +968,265 @@ impl WsFederationApi<WsClient> {
+                let mut backoff = api_networking_backoff();
+
+                while let Ok(sender) = receiver.recv().await {
+                    let n_request = receiver.len();
+
+                    match connector.connect(peer).await {
+                        Ok(connection) => {
+                            info!(target: LOG_CLIENT_NET_API, "Connected to peer api");
+
+                            sender.send(connection.clone()).ok();
+
+                            loop {
+                                tokio::select! {
+                                    sender = receiver.recv() => {
+                                        match sender.ok() {
+                                            Some(sender) => sender.send(connection.clone()).ok(),
+                                            None => break,
+                                        };
+                                    }
+                                    () = connection.await_disconnection() => break,
+                                }
+                            }
+
+                            info!(target: LOG_CLIENT_NET_API, "Disconnected from peer api");
+
+                            backoff = api_networking_backoff();
+                        }
+                        Err(e) => {
+                            info!(target: LOG_CLIENT_NET_API, "Failed to connect to peer api {e}");
+
+                            // We need to drain the channel for the pending requests to fail.
+
+                            for _ in 0..n_request {
+                                receiver.try_recv().expect("Items exist");
+                            }
+
+                            fedimint_core::task::sleep(
+                                backoff.next().expect("No limit to the number of retries"),
+                            )
+                            .await;
+                        }
                     }
-                    debug!(target: LOG_CLIENT_NET_API, "Triggering reconnection after disconnection");
                 }
-            };
 
-            // Drop read lock so we can take the write lock, which is needed to reconnect.
-            drop(rclient);
-            let mut wclient = self.client.write().await;
-            match wclient.client.get_try().await {
-                Ok(client) if client.is_connected() => {
-                    // someone else connected, just loop again
-                    trace!(target: LOG_CLIENT_NET_API, "Some other request reconnected client, retrying");
-                }
-                _ => {
-                    wclient.reconnect(
-                        self.connector,
-                        self.peer_id,
-                        self.url.clone(),
-                        self.api_secret.clone(),
-                    );
-                }
+                info!(target: LOG_CLIENT_NET_API, "Shutting down peer api connection task");
             }
+            .instrument(info_span!("peer-api-connection", ?peer)),
```

Comment:
This one is extremely noisy.

![image](https://github.com/user-attachments/assets/e089eaf3-ad6f-4dcd-9e98-742fea96a2b9)